async-profiler-context/src/main/java/io/pyroscope/labels/ScopedContext.java

```diff
@@ -5,19 +5,21 @@
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.BiConsumer;
 
 public class ScopedContext implements AutoCloseable {
-    static final ThreadLocal<Context> context = ThreadLocal.withInitial(() ->
-            new Context(0L, Collections.emptyMap())
-    );
+    static final Map<Thread, Context> threadContext = new ConcurrentHashMap<>();
```

Comment:
Does this leak a `Thread` object when the thread exits?

Reply:
I have changed to WeakHashMap and added a test to guarantee it is not leaking.